collections/binary_search_tree.ts

```diff
@@ -188,11 +188,6 @@ export class BinarySearchTree<T> implements Iterable<T> {
     const replacementDirection: Direction = direction === "left"
       ? "right"
       : "left";
-    if (!node[replacementDirection]) {
```

Comment:
I think this check is still useful for the user who extends this class. You can test this validation by the code like below:

```ts
import { BinarySearchTree  } from "./collections/binary_search_tree.ts";
class MyTree<T> extends BinarySearchTree<T> {
  foo() {
    super.rotateNode(this.root!, "right");
  }
}
const tree = new MyTree();
tree.insert(1);
tree.foo();
```

Reply:
I see. Yes, it throws a useful error. I reverted and actually used this snippet as a test instead.